projects/storefrontlib/router/on-navigate.service.ts

```diff
@@ -33,8 +27,10 @@ export class OnNavigateService {
 
   protected subscription: Subscription;
 
-  get hostComponent(): ComponentRef<any> {
-    return this.injector.get(ApplicationRef)?.components?.[0];
+  get hostComponent(): HTMLElement | undefined {
+    return <HTMLElement>(
+      this.injector.get(DOCUMENT).getElementsByTagName('cx-storefront')?.[0]
```

Comment:
Perhaps we can make the root component tag definable in config. It can be introduced as an additional feature rather than a breaking change.